common/jmeter/perf-test-pre-provisioned.sh

```
@@ -567,67 +567,125 @@ function test_scenarios() {
-
-                echo ""
-                echo "Zipping JTL files in $report_location"
-                zip -jm "$report_location"/jtls.zip "$report_location"/results*.jtl
+            local scenario_mode=${scenario[modes]:-""}
+
+            # Determine org count sweep: B2B scenarios iterate over org counts,
+            # all other scenarios use a single-element sentinel array so the
+            # inner loop body runs exactly once with no org-count JMeter param.
+            local -a org_counts_for_scenario
+            if [ "$scenario_mode" = "B2B" ]; then
+                org_counts_for_scenario=("${b2b_org_counts_array[@]}")
```

Shellcheck (0.11.0)

[warning] b2b_org_counts_array is referenced but not assigned.

(SC2154)